server/src/main/java/com/thoughtworks/go/config/ConfigConverter.java

```diff
@@ -639,4 +641,395 @@ private TrackingTool toTrackingTool(CRTrackingTool crTrackingTool) {
+        return crMingle;
+    }
+
+    private CRTrackingTool trackingToolToCRTrackingTool(TrackingTool trackingTool) {
+        if (trackingTool == null)
+            return null;
+        return new CRTrackingTool(trackingTool.getLink(), trackingTool.getRegex());
+    }
+
+    private CRParameter paramToCRParam(ParamConfig paramConfig) {
+        return new CRParameter(paramConfig.getName(), paramConfig.getValue());
+    }
+
+    public CRTimer timerConfigToCRTimer(TimerConfig timerConfig) {
+        if (timerConfig == null)
+            return null;
+        String spec =  timerConfig.getTimerSpec();
+        if (StringUtils.isBlank(spec))
+            throw new RuntimeException("timer schedule is not specified");
+        return new CRTimer(spec, timerConfig.shouldTriggerOnlyOnChanges());
+    }
+
+    public CREnvironmentVariable environmentVariableConfigToCREnvironmentVariable(EnvironmentVariableConfig environmentVariableConfig) {
+        if (environmentVariableConfig.isSecure()) {
+            return new CREnvironmentVariable(environmentVariableConfig.getName(), null, environmentVariableConfig.getEncryptedValue());
+        } else {
+            String value = environmentVariableConfig.getValue();
+            if(StringUtils.isBlank(value))
+                value = "";
+            return new CREnvironmentVariable(environmentVariableConfig.getName(), value);
+        }
+    }
+
+    private CRPluggableScmMaterial pluggableScmMaterialConfigToCRPluggableScmMaterial(PluggableSCMMaterialConfig pluggableScmMaterialConfig) {
+        SCMs scms = getSCMs();
+        String id = pluggableScmMaterialConfig.getScmId();
+        SCM scmConfig = scms.find(id);
+        if (scmConfig == null)
+            throw new ConfigConvertionException(
+                    String.format("Failed to find referenced scm '%s'", id));
+
+        return new CRPluggableScmMaterial(pluggableScmMaterialConfig.getName().toString(),
+                id, pluggableScmMaterialConfig.getFolder(),
+                pluggableScmMaterialConfig.getFilterAsString());
+    }
+
+    public CRPackageMaterial packageMaterialToCRPackageMaterial(PackageMaterialConfig packageMaterialConfig) {
+        return new CRPackageMaterial(packageMaterialConfig.getName().toString(), packageMaterialConfig.getPackageId());
+    }
+
+    public CRDependencyMaterial dependencyMaterialConfigToCRDependencyMaterial(DependencyMaterialConfig dependencyMaterialConfig) {
+        CRDependencyMaterial crDependencyMaterial = new CRDependencyMaterial(
+                dependencyMaterialConfig.getPipelineName().toString(),
+                dependencyMaterialConfig.getStageName().toString());
+        if (dependencyMaterialConfig.getName() != null)
+            crDependencyMaterial.setName(dependencyMaterialConfig.getName().toString());
+        return crDependencyMaterial;
+    }
+
+    private CRScmMaterial scmMaterialToCRScmMaterial(ScmMaterialConfig scmMaterialConfig) {
```

Comment:
Can you break this method up a bit, at least? It's hard to follow. I know the original was similar.  :) But, I think this can be rewritten a bit, even if it's a little longer.

Reply:
Yeah... It's pretty cumbersome. I'll break it up a bit

Reply:
@ibnc Looks alright to me.